starcheck/src/lib/Ska/Starcheck/Obsid.pm

```diff
@@ -21,6 +21,53 @@ package Ska::Starcheck::Obsid;
 use strict;
 use warnings;
 
+use Inline Python => q{
+
+from astropy.table import Table, Column
+import Quaternion
+from Ska.quatutil import radec2yagzag
+import agasc
+
+
+def _get_agasc_stars(ra, dec, roll, radius, date, agasc_file):
+    """
+    Fetch the cone of agasc stars.  Update the table with the yag and zag of each star.
+    Return as a dictionary with the agasc ids as keys and all of the values as
+    simple Python types (int, float)
+    """
+    stars = agasc.get_agasc_cone(float(ra), float(dec), float(radius), date, agasc_file)
+    q_aca = Quaternion.Quat([float(ra), float(dec), float(roll)])
+    yags, zags = radec2yagzag(stars['RA_PMCORR'], stars['DEC_PMCORR'], q_aca)
+    yags *= 3600
+    zags *= 3600
+    stars['yang'] = yags
+    stars['zang'] = zags
+
+    # Get a dictionary of the stars with the columns that are used
+    # This needs to be de-numpy-ified to pass back into Perl
+    stars_dict = {}
+    for star in stars:
+        stars_dict["{}".format(star['AGASC_ID'])] = {
+            'id': int(star['AGASC_ID']),
```

Comment:
This could get cleaned up with something that just matches the columns to functions to convert to plain Python types, but there weren't that many columns (to just write it out this way).

Reply:
FYI you can use `.item()` on a numpy object to convert to the appropriate native Python type.  I believe `tolist()` would also work here.  It won't shorten this code much and maybe having the explicit types is fine anyway.